<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">
<head>
  <meta charset="utf-8" />
  <meta name="generator" content="pandoc" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
  <title>snpvar_part1</title>
  <style type="text/css">
      code{white-space: pre-wrap;}
      span.smallcaps{font-variant: small-caps;}
      span.underline{text-decoration: underline;}
      div.column{display: inline-block; vertical-align: top; width: 50%;}
  </style>
  <style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    background-color: #ffffff;
    color: #a0a0a0;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #a0a0a0;  padding-left: 4px; }
div.sourceCode
  { color: #1f1c1b; background-color: #ffffff; }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span. { color: #1f1c1b; } /* Normal */
code span.al { color: #bf0303; background-color: #f7e6e6; font-weight: bold; } /* Alert */
code span.an { color: #ca60ca; } /* Annotation */
code span.at { color: #0057ae; } /* Attribute */
code span.bn { color: #b08000; } /* BaseN */
code span.bu { color: #644a9b; font-weight: bold; } /* BuiltIn */
code span.cf { color: #1f1c1b; font-weight: bold; } /* ControlFlow */
code span.ch { color: #924c9d; } /* Char */
code span.cn { color: #aa5500; } /* Constant */
code span.co { color: #898887; } /* Comment */
code span.cv { color: #0095ff; } /* CommentVar */
code span.do { color: #607880; } /* Documentation */
code span.dt { color: #0057ae; } /* DataType */
code span.dv { color: #b08000; } /* DecVal */
code span.er { color: #bf0303; text-decoration: underline; } /* Error */
code span.ex { color: #0095ff; font-weight: bold; } /* Extension */
code span.fl { color: #b08000; } /* Float */
code span.fu { color: #644a9b; } /* Function */
code span.im { color: #ff5500; } /* Import */
code span.in { color: #b08000; } /* Information */
code span.kw { color: #1f1c1b; font-weight: bold; } /* Keyword */
code span.op { color: #1f1c1b; } /* Operator */
code span.ot { color: #006e28; } /* Other */
code span.pp { color: #006e28; } /* Preprocessor */
code span.re { color: #0057ae; background-color: #e0e9f8; } /* RegionMarker */
code span.sc { color: #3daee9; } /* SpecialChar */
code span.ss { color: #ff5500; } /* SpecialString */
code span.st { color: #bf0303; } /* String */
code span.va { color: #0057ae; } /* Variable */
code span.vs { color: #bf0303; } /* VerbatimString */
code span.wa { color: #bf0303; } /* Warning */
  </style>
  <link rel="stylesheet" href="avenir-white.css" />
  <!--[if lt IE 9]>
    <script src="//cdnjs.cloudflare.com/ajax/libs/html5shiv/3.7.3/html5shiv-printshiv.min.js"></script>
  <![endif]-->
</head>
<body>
<h1 id="human-population-genomic-variation">Human Population Genomic Variation</h1>
<h2 id="variant-snp-rs1426654-in-slc24a5-gene-part-1">Variant (SNP rs1426654) in <em>SLC24A5</em> Gene: Part 1</h2>
<h3 id="molb-44855485-computers-in-biology">Molb 4485/5485 – Computers in Biology</h3>
<h3 id="vikram-chhatre-and-nicolas-blouin">Vikram Chhatre and Nicolas Blouin</h3>
<p>Wyoming INBRE Bioinformatics Core<br />
Dept. of Molecular Biology<br />
University of Wyoming</p>
<a href="mailto:vchhatre@uwyo.edu">vchhatre@uwyo.edu</a><br />
<a href="mailto:nblouin@uwyo.edu">nblouin@uwyo.edu</a><br />
<a href="http://molb4485.uwyo.online" class="uri">http://molb4485.uwyo.online</a><br />

<hr>
<p><img src="2001.jpg" width=800></a> The Dawn of Man. Image Credit: <a href="http://moviemezzanine.com">Movie Mezannine</a></p>
<h3 style="color: red;">
Important:
</h3>
<p><strong>1. Put away your phone for the duration of the class</strong></p>
<p><strong>2. Do not race ahead in the tutorial.</strong></p>
<p><strong>3. Doing so compromises your learning experience.</strong></p>
<p><strong>4. You may not leave early today!</strong> <br><br><br></p>
<h2 id="table-of-contents">Table of Contents</h2>
<ol type="1">
<li><a href="#data">Exercise Data</a></li>
<li><a href="#vcf">Variant Call Format</a></li>
<li><a href="#freq">Gene Frequency</a></li>
</ol>
<p><br><br><br><br><br></p>
<p><a name="data"></a></p>
<h3 id="data-for-todays-exercise">1. Data For Today’s Exercise</h3>
<p>For this entire exercise, replace <code>&lt;inbreNNN&gt;</code> with your own <code>inbre</code> ID. Once you log onto the Teton computing environment, copy the data for today’s exercise as follows.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb1-1" data-line-number="1">$ <span class="fu">ssh</span> <span class="op">&lt;</span>inbreNNN<span class="op">&gt;</span>@teton.uwyo.edu</a>
<a class="sourceLine" id="cb1-2" data-line-number="2"></a>
<a class="sourceLine" id="cb1-3" data-line-number="3">$ <span class="bu">cd</span> /project/inbre-train/<span class="op">&lt;</span>inbreNNN<span class="op">&gt;</span>/</a>
<a class="sourceLine" id="cb1-4" data-line-number="4"></a>
<a class="sourceLine" id="cb1-5" data-line-number="5">$ <span class="fu">mkdir</span> Week5_LastName <span class="kw">&amp;&amp;</span> <span class="bu">cd</span> Week5_LastName/</a>
<a class="sourceLine" id="cb1-6" data-line-number="6"></a>
<a class="sourceLine" id="cb1-7" data-line-number="7">$ <span class="fu">cp</span> -r /project/inbre-train/Week5Data/* .</a>
<a class="sourceLine" id="cb1-8" data-line-number="8"></a>
<a class="sourceLine" id="cb1-9" data-line-number="9">$ <span class="fu">ls</span> -lh</a>
<a class="sourceLine" id="cb1-10" data-line-number="10"></a></code></pre></div>
<p><a name="vcf"></a></p>
<h3 id="variant-call-format-vcf-spec.-4.1">2. Variant Call Format (VCF) spec. 4.1</h3>
<p>This is a standard format for storing information on polymorphism across the genome. It is a <code>n x n</code> matrix of individuals and variants. Columns refer to individuals, whereas rows refer to variants (SNPs). Analogous to the FASTQ format, VCF also stores quality information. Files formatted to store variant calls always end in <code>.vcf</code> extension. For example: <code>input.vcf</code>. The VCF format specification is described in detail in the user manual available at <a href="http://samtools.github.io" class="uri">http://samtools.github.io</a>.</p>
<h4 id="vcftools">2.1 VCFTools</h4>
<p>VCFTools is a program that can parse and manipulate the information in <code>.vcf</code> files. A comprehensive user manual on VCFTools usage is <a href="http://vcftools.sourceforge.net/man%20latest.html"><strong>available here</strong></a>. Before we can use this program, we must load it in memory as a module.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="ex">module</span> load gcc swset</a>
<a class="sourceLine" id="cb2-2" data-line-number="2"></a>
<a class="sourceLine" id="cb2-3" data-line-number="3"><span class="ex">module</span> load vcftools</a>
<a class="sourceLine" id="cb2-4" data-line-number="4"></a>
<a class="sourceLine" id="cb2-5" data-line-number="5"><span class="ex">vcftools</span></a>
<a class="sourceLine" id="cb2-6" data-line-number="6"></a>
<a class="sourceLine" id="cb2-7" data-line-number="7"><span class="ex">VCFtools</span> (0.1.14)</a>
<a class="sourceLine" id="cb2-8" data-line-number="8">© <span class="ex">Adam</span> Auton and Anthony Marcketta 2009</a>
<a class="sourceLine" id="cb2-9" data-line-number="9"></a>
<a class="sourceLine" id="cb2-10" data-line-number="10"><span class="ex">Process</span> Variant Call Format files</a>
<a class="sourceLine" id="cb2-11" data-line-number="11"></a>
<a class="sourceLine" id="cb2-12" data-line-number="12"><span class="ex">For</span> a list of options, please go to:</a>
<a class="sourceLine" id="cb2-13" data-line-number="13">    <span class="ex">https</span>://vcftools.github.io/man_latest.html</a>
<a class="sourceLine" id="cb2-14" data-line-number="14"></a>
<a class="sourceLine" id="cb2-15" data-line-number="15"><span class="ex">Alternatively</span>, a man page is available, type:</a>
<a class="sourceLine" id="cb2-16" data-line-number="16">    <span class="fu">man</span> vcftools</a>
<a class="sourceLine" id="cb2-17" data-line-number="17"></a>
<a class="sourceLine" id="cb2-18" data-line-number="18"><span class="ex">Questions</span>, comments, and suggestions should be emailed to:</a>
<a class="sourceLine" id="cb2-19" data-line-number="19">    <span class="ex">vcftools-help@lists.sourceforge.net</span></a>
<a class="sourceLine" id="cb2-20" data-line-number="20"></a></code></pre></div>
<h4 id="vcf-file">2.2 VCF File</h4>
<p>We have provided you with an example <code>.vcf</code> file for today’s exercise: <code>human_chr15.vcf</code>. As the name suggests, this file contains data from <em>Homo sapiens</em> <strong>Chromosome 15</strong>. The data was collected as part of the <a href="http://www.internationalgenome.org">1000 Genomes Project</a> whose aim is to understand human genomic variation across the globe. This data is from the final phase of the project and contains genotypes of over 2500 individuals from 26 populations distributed globally genotyped at more than 84 million SNPs. See the map below for where those populations come from:</p>
<center>
<img src='pops.png' width=800> </img> Image Credit: <a href="http://internationalgenome.org">1000 Genomes Project</a>
</center>
<p>To make it more manageable, we have subset the original data to only include information on 16000 SNPs. You will be further subsetting this file to obtain data on one specific SNP that we are interested in. But for now, open the file in text editor and take a look at the contents. Then close the file without saving.</p>
<p><strong>Note</strong>: <code>vi</code> that you have been using so far is synonymous with <code>vim</code>. Both commands work the same.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb3-1" data-line-number="1"></a>
<a class="sourceLine" id="cb3-2" data-line-number="2"><span class="ex">vim</span> human_chr15.vcf</a>
<a class="sourceLine" id="cb3-3" data-line-number="3"></a>
<a class="sourceLine" id="cb3-4" data-line-number="4">[<span class="ex">vim</span>] :set nowrap  ## This helps with unwrapping the lines.</a>
<a class="sourceLine" id="cb3-5" data-line-number="5">[<span class="ex">vim</span>] :wq</a></code></pre></div>
<h4 id="vcf-data-summary">2.3 VCF Data Summary</h4>
<p>VCFTools can provide a quick summary of your data. When running without any options, VCFTools outputs summary statistics but does not make any changes to the file nor does it produce any output files.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb4-1" data-line-number="1"></a>
<a class="sourceLine" id="cb4-2" data-line-number="2"><span class="ex">vcftools</span> --vcf human_chr15.vcf </a>
<a class="sourceLine" id="cb4-3" data-line-number="3"></a>
<a class="sourceLine" id="cb4-4" data-line-number="4"><span class="ex">VCFtools</span> - 0.1.14</a>
<a class="sourceLine" id="cb4-5" data-line-number="5"><span class="kw">(</span><span class="ex">C</span><span class="kw">)</span> <span class="ex">Adam</span> Auton and Anthony Marcketta 2009</a>
<a class="sourceLine" id="cb4-6" data-line-number="6"></a>
<a class="sourceLine" id="cb4-7" data-line-number="7"><span class="ex">Parameters</span> as interpreted:</a>
<a class="sourceLine" id="cb4-8" data-line-number="8">    <span class="ex">--vcf</span> human_chr15.vcf</a>
<a class="sourceLine" id="cb4-9" data-line-number="9"></a>
<a class="sourceLine" id="cb4-10" data-line-number="10"><span class="ex">After</span> filtering, kept 2504 out of 2504 Individuals</a>
<a class="sourceLine" id="cb4-11" data-line-number="11"><span class="ex">After</span> filtering, kept 16003 out of a possible 16003 Sites</a>
<a class="sourceLine" id="cb4-12" data-line-number="12"><span class="ex">Run</span> Time = 1.00 seconds</a></code></pre></div>
<h4 id="snp-rs1426654">2.4 SNP <code>rs1426654</code></h4>
<p>For today’s exercise, we are interested in only one SNP: <code>rs1426654</code>. Every documented SNP from the human genome has a <code>rs</code> ID associated with it. Note that the number following rs, is not the physical position of this SNP on a chromosome, but simply an identifier. You can look up the physical position of this SNP in the VCF file. For example:</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb5-1" data-line-number="1"></a>
<a class="sourceLine" id="cb5-2" data-line-number="2"><span class="fu">grep</span> <span class="st">&quot;rs1426654&quot;</span> human_chr15.vcf <span class="kw">|</span> <span class="fu">cut</span> -f1-5</a>
<a class="sourceLine" id="cb5-3" data-line-number="3"></a>
<a class="sourceLine" id="cb5-4" data-line-number="4"><span class="ex">15</span>  48426484    rs1426654   A   G</a></code></pre></div>
<p>Why did we pass the output of grep to cut via a <code>|</code>? It’s because when grep matches a pattern, it prints the entire line to the screen. Our lines here have more than 2500 columns and at this time, we are only interested in the first few columns as displayed above.</p>
<p>The gene <strong>SLC24A5</strong> encodes a protein called <strong>Na/K/Ca Exchanger 5</strong> which greatly influences skin color in humans. The leading hypothesis is that when our ancestors traveled out of Africa into Europe, the darker skin color was of no advantage in Europe’s cold climate. So eventually a new variant of the gene evolved in northern human populations which reduced the amount of skin pigmentation. By comparing the frequency of the two forms (alleles) of this gene in various human populations, we can study how populations can adapt differently when they are subject to very different environments.</p>
<h4 id="subset-vcf-for-rs1426654">2.5 Subset VCF for <code>rs1426654</code></h4>
<p>While there are more than 16,000 SNPs in our vcf file, we only need one for today’s exercise. So let’s go ahead and subset the VCF for that SNP.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb6-1" data-line-number="1"><span class="ex">vcftools</span> --vcf human_chr15.vcf --snp rs1426654 --recode --out SLC24A5</a>
<a class="sourceLine" id="cb6-2" data-line-number="2"></a>
<a class="sourceLine" id="cb6-3" data-line-number="3"></a>
<a class="sourceLine" id="cb6-4" data-line-number="4"><span class="ex">VCFtools</span> - 0.1.14</a>
<a class="sourceLine" id="cb6-5" data-line-number="5"><span class="kw">(</span><span class="ex">C</span><span class="kw">)</span> <span class="ex">Adam</span> Auton and Anthony Marcketta 2009</a>
<a class="sourceLine" id="cb6-6" data-line-number="6"></a>
<a class="sourceLine" id="cb6-7" data-line-number="7"><span class="ex">Parameters</span> as interpreted:</a>
<a class="sourceLine" id="cb6-8" data-line-number="8">    <span class="ex">--vcf</span> human_chr15.vcf</a>
<a class="sourceLine" id="cb6-9" data-line-number="9">    <span class="ex">--out</span> SLC24A5</a>
<a class="sourceLine" id="cb6-10" data-line-number="10">    <span class="ex">--recode</span></a>
<a class="sourceLine" id="cb6-11" data-line-number="11">    <span class="ex">--snp</span> rs1426654</a>
<a class="sourceLine" id="cb6-12" data-line-number="12"></a>
<a class="sourceLine" id="cb6-13" data-line-number="13"><span class="ex">After</span> filtering, kept 2504 out of 2504 Individuals</a>
<a class="sourceLine" id="cb6-14" data-line-number="14"><span class="ex">Outputting</span> VCF file...</a>
<a class="sourceLine" id="cb6-15" data-line-number="15"><span class="ex">After</span> filtering, kept 1 out of a possible 16003 Sites</a>
<a class="sourceLine" id="cb6-16" data-line-number="16"><span class="ex">Run</span> Time = 1.00 seconds</a></code></pre></div>
<p>In the above command, we extracted data for SNP rs1426654 from the file human_chr15.vcf and provided a prefix <code>SLC24A5</code> for the output file. Check the output:</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb7-1" data-line-number="1"></a>
<a class="sourceLine" id="cb7-2" data-line-number="2"><span class="fu">ls</span> -lh</a>
<a class="sourceLine" id="cb7-3" data-line-number="3"></a>
<a class="sourceLine" id="cb7-4" data-line-number="4"><span class="ex">-rw-rw-r--</span> 1 vchhatre inbre-train 156M Nov  5 21:44 human_chr15.vcf</a>
<a class="sourceLine" id="cb7-5" data-line-number="5"><span class="ex">-rw-rw-r--</span> 1 vchhatre inbre-train  307 Nov  5 22:54 SLC24A5.log</a>
<a class="sourceLine" id="cb7-6" data-line-number="6"><span class="ex">-rw-rw-r--</span> 1 vchhatre inbre-train  30K Nov  5 22:54 SLC24A5.recode.vcf</a></code></pre></div>
<p>The information that was printed to the screen was also populated inside the log file <code>SLC24A5.log</code>. Your main output file is <code>SLC24A5.recode.vcf</code>, but let’s change its name to <code>SLC24A5.vcf</code> for convenience.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb8-1" data-line-number="1"></a>
<a class="sourceLine" id="cb8-2" data-line-number="2"></a>
<a class="sourceLine" id="cb8-3" data-line-number="3">$ <span class="fu">mv</span> SLC24A5.recode.vcf SLC24A5.vcf</a>
<a class="sourceLine" id="cb8-4" data-line-number="4"></a>
<a class="sourceLine" id="cb8-5" data-line-number="5">$ <span class="fu">ls</span> -lh</a>
<a class="sourceLine" id="cb8-6" data-line-number="6"></a>
<a class="sourceLine" id="cb8-7" data-line-number="7"><span class="ex">-rw-rw-r--</span> 1 vchhatre inbre-train  30K Nov  5 22:54 SLC24A5.recode.vcf</a></code></pre></div>
<p><a name="freq"></a></p>
<h2 id="gene-frequency">3. Gene Frequency</h2>
<p>As we discussed earlier, allele frequency is a measure of the relative proportion of the two (<em>sometimes more</em>) forms of a gene in a population. If a given SNP is not polymorphic, it will have only one allele (<em>form</em>). Frequency of that form will be <code>f = 1.00</code> or 100%. However, if it is polymorphic, the resulting gene forms (alleles) may be present at any frquency from 0.01 to 0.99, whereas the other allele’s frequency will then be proportionately between 0.99 and 0.1. No matter their relative proportion, the frequencies of the two alleles will always add up to 1 in any population sample.</p>
<h3 id="frequency-of-rs1426654-alleles">3.1 Frequency of <code>rs1426654</code> Alleles</h3>
<p>We can use VCFTools to quickly estimate frequencies of the two alleles (<code>A/G</code>). But remember that allele frequency is a measure of population(s). During today’s exercise, we want to calculate these frequencies in each of the 26 human populations that we have data for. However, the vcf files does not contain any information on which individuals belong to which populations. Instead the population classification information is present in the folder <code>popfiles</code>:</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb9-1" data-line-number="1"></a>
<a class="sourceLine" id="cb9-2" data-line-number="2">$ <span class="fu">ls</span> -lh popfiles/</a>
<a class="sourceLine" id="cb9-3" data-line-number="3"><span class="ex">total</span> 0</a>
<a class="sourceLine" id="cb9-4" data-line-number="4"></a>
<a class="sourceLine" id="cb9-5" data-line-number="5"><span class="ex">-rw-r--r--</span> 1 vchhatre  792 Nov  5 21:55 ESN.pop</a>
<a class="sourceLine" id="cb9-6" data-line-number="6"><span class="ex">-rw-r--r--</span> 1 vchhatre  816 Nov  5 21:55 STU.pop</a>
<a class="sourceLine" id="cb9-7" data-line-number="7"><span class="ex">-rw-r--r--</span> 1 vchhatre  768 Nov  5 21:55 PJL.pop</a>
<a class="sourceLine" id="cb9-8" data-line-number="8"><span class="ex">-rw-r--r--</span> 1 vchhatre  856 Nov  5 21:55 IBS.pop</a>
<a class="sourceLine" id="cb9-9" data-line-number="9"><span class="ex">-rw-r--r--</span> 1 vchhatre  744 Nov  5 21:55 CDX.pop</a>
<a class="sourceLine" id="cb9-10" data-line-number="10"><span class="ex">-rw-r--r--</span> 1 vchhatre  752 Nov  5 21:55 CLM.pop</a>
<a class="sourceLine" id="cb9-11" data-line-number="11"><span class="ex">-rw-r--r--</span> 1 vchhatre  512 Nov  5 21:55 MXL.pop</a>
<a class="sourceLine" id="cb9-12" data-line-number="12"><span class="ex">-rw-r--r--</span> 1 vchhatre  824 Nov  5 21:55 CHB.pop</a>
<a class="sourceLine" id="cb9-13" data-line-number="13"><span class="ex">-rw-r--r--</span> 1 vchhatre  832 Nov  5 21:55 PUR.pop</a>
<a class="sourceLine" id="cb9-14" data-line-number="14"><span class="ex">-rw-r--r--</span> 1 vchhatre  792 Nov  5 21:55 KHV.pop</a>
<a class="sourceLine" id="cb9-15" data-line-number="15"><span class="ex">-rw-r--r--</span> 1 vchhatre  824 Nov  5 21:55 GIH.pop</a>
<a class="sourceLine" id="cb9-16" data-line-number="16"><span class="ex">-rw-r--r--</span> 1 vchhatre  680 Nov  5 21:55 PEL.pop</a>
<a class="sourceLine" id="cb9-17" data-line-number="17"><span class="ex">-rw-r--r--</span> 1 vchhatre  728 Nov  5 21:55 GBR.pop</a>
<a class="sourceLine" id="cb9-18" data-line-number="18"><span class="ex">-rw-r--r--</span> 1 vchhatre  856 Nov  5 21:55 TSI.pop</a>
<a class="sourceLine" id="cb9-19" data-line-number="19"><span class="ex">-rw-r--r--</span> 1 vchhatre  792 Nov  5 21:55 LWK.pop</a>
<a class="sourceLine" id="cb9-20" data-line-number="20"><span class="ex">-rw-r--r--</span> 1 vchhatre  680 Nov  5 21:55 MSL.pop</a>
<a class="sourceLine" id="cb9-21" data-line-number="21"><span class="ex">-rw-r--r--</span> 1 vchhatre  688 Nov  5 21:55 BEB.pop</a>
<a class="sourceLine" id="cb9-22" data-line-number="22"><span class="ex">-rw-r--r--</span> 1 vchhatre  864 Nov  5 21:55 YRI.pop</a>
<a class="sourceLine" id="cb9-23" data-line-number="23"><span class="ex">-rw-r--r--</span> 1 vchhatre  840 Nov  5 21:55 CHS.pop</a>
<a class="sourceLine" id="cb9-24" data-line-number="24"><span class="ex">-rw-r--r--</span> 1 vchhatre  816 Nov  5 21:55 ITU.pop</a>
<a class="sourceLine" id="cb9-25" data-line-number="25"><span class="ex">-rw-r--r--</span> 1 vchhatre  792 Nov  5 21:55 FIN.pop</a>
<a class="sourceLine" id="cb9-26" data-line-number="26"><span class="ex">-rw-r--r--</span> 1 vchhatre  832 Nov  5 21:55 JPT.pop</a>
<a class="sourceLine" id="cb9-27" data-line-number="27"><span class="ex">-rw-r--r--</span> 1 vchhatre  488 Nov  5 21:55 ASW.pop</a>
<a class="sourceLine" id="cb9-28" data-line-number="28"><span class="ex">-rw-r--r--</span> 1 vchhatre  792 Nov  5 21:55 CEU.pop</a>
<a class="sourceLine" id="cb9-29" data-line-number="29"><span class="ex">-rw-r--r--</span> 1 vchhatre  904 Nov  5 21:55 GWD.pop</a>
<a class="sourceLine" id="cb9-30" data-line-number="30"><span class="ex">-rw-r--r--</span> 1 vchhatre  768 Nov  5 21:55 ACB.pop</a></code></pre></div>
<p>Each of these popfiles contains a list of individuals that belong to that specific population. Try using the linux command <code>cat</code> on some of the files to see what is inside. You will notice that number of lines in each file corresponds to the number of individuals in it, with one individual listed per line.</p>
<p>Next, we will be writing a simple shell script to automate the estimation of allele frequencies using these population files and VCFTools. This script consumes very little computational resources, so we will not run it using sbatch. Thus you can exclude the scheduler commands that begin with <code>#SBATCH</code> for this script.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb10-1" data-line-number="1"></a>
<a class="sourceLine" id="cb10-2" data-line-number="2"><span class="ex">vim</span> makefreq.sh</a>
<a class="sourceLine" id="cb10-3" data-line-number="3"></a>
<a class="sourceLine" id="cb10-4" data-line-number="4">   <span class="co">#!/bin/bash</span></a>
<a class="sourceLine" id="cb10-5" data-line-number="5"></a>
<a class="sourceLine" id="cb10-6" data-line-number="6">   <span class="co">## First load the modules</span></a>
<a class="sourceLine" id="cb10-7" data-line-number="7">   <span class="ex">module</span> load gcc swset vcftools</a>
<a class="sourceLine" id="cb10-8" data-line-number="8"></a>
<a class="sourceLine" id="cb10-9" data-line-number="9">   <span class="co">## Move to the appropriate directory</span></a>
<a class="sourceLine" id="cb10-10" data-line-number="10">   <span class="bu">cd</span> /project/inbre-train/<span class="op">&lt;</span>inbreNNN<span class="op">&gt;</span>/Week5_LastName/</a>
<a class="sourceLine" id="cb10-11" data-line-number="11"></a>
<a class="sourceLine" id="cb10-12" data-line-number="12">   <span class="co">## Make folder to store frequency output</span></a>
<a class="sourceLine" id="cb10-13" data-line-number="13">   <span class="fu">mkdir</span> freq</a>
<a class="sourceLine" id="cb10-14" data-line-number="14"></a>
<a class="sourceLine" id="cb10-15" data-line-number="15">   <span class="co">## Navigate to the folder ‘popfiles/’</span></a>
<a class="sourceLine" id="cb10-16" data-line-number="16">   <span class="bu">cd</span> popfiles/</a>
<a class="sourceLine" id="cb10-17" data-line-number="17"></a>
<a class="sourceLine" id="cb10-18" data-line-number="18">   <span class="co">## For loop</span></a>
<a class="sourceLine" id="cb10-19" data-line-number="19">   <span class="kw">for</span> <span class="ex">p</span> in *.pop</a>
<a class="sourceLine" id="cb10-20" data-line-number="20">   <span class="kw">do</span></a>
<a class="sourceLine" id="cb10-21" data-line-number="21">     <span class="ex">vcftools</span> --vcf ../SLC24A5.vcf --keep <span class="va">$p</span> --freq --out ../freq/<span class="va">$p</span></a>
<a class="sourceLine" id="cb10-22" data-line-number="22">   <span class="kw">done</span></a>
<a class="sourceLine" id="cb10-23" data-line-number="23">   <span class="bu">cd</span> ..</a>
<a class="sourceLine" id="cb10-24" data-line-number="24"></a></code></pre></div>
<p>Save and close the file. This script demonstrates a simple <code>for</code> loop which iterates over popfiles shown above and executes vcftools (to estimate allele frequencies) for only the individuals in the population file currently under iteration (Think back to our example of two populations during the lecture). Go ahead and execute it. Then look at the results.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb11-1" data-line-number="1"></a>
<a class="sourceLine" id="cb11-2" data-line-number="2"><span class="fu">bash</span> makefreq.sh  ## Note, it is <span class="st">&#39;bash&#39;</span>, not <span class="st">&#39;sbatch&#39;</span>.</a>
<a class="sourceLine" id="cb11-3" data-line-number="3"></a>
<a class="sourceLine" id="cb11-4" data-line-number="4"><span class="bu">pwd</span></a>
<a class="sourceLine" id="cb11-5" data-line-number="5"><span class="ex">/project/inbre-train</span>/<span class="op">&lt;</span><span class="ex">inbreNNN</span><span class="op">&gt;</span>/Week5_LastName</a>
<a class="sourceLine" id="cb11-6" data-line-number="6"></a>
<a class="sourceLine" id="cb11-7" data-line-number="7"><span class="bu">cd</span> freq</a>
<a class="sourceLine" id="cb11-8" data-line-number="8"></a>
<a class="sourceLine" id="cb11-9" data-line-number="9"><span class="fu">ls</span> -othr *.frq</a>
<a class="sourceLine" id="cb11-10" data-line-number="10"></a>
<a class="sourceLine" id="cb11-11" data-line-number="11"><span class="ex">-rw-rw-r--</span> 1 vchhatre 80 Oct 28 23:55 ACB.pop.frq</a>
<a class="sourceLine" id="cb11-12" data-line-number="12"><span class="ex">-rw-rw-r--</span> 1 vchhatre 80 Oct 28 23:55 ASW.pop.frq</a>
<a class="sourceLine" id="cb11-13" data-line-number="13"><span class="ex">-rw-rw-r--</span> 1 vchhatre 80 Oct 28 23:55 BEB.pop.frq</a>
<a class="sourceLine" id="cb11-14" data-line-number="14"><span class="ex">-rw-rw-r--</span> 1 vchhatre 66 Oct 28 23:55 CDX.pop.frq</a>
<a class="sourceLine" id="cb11-15" data-line-number="15"><span class="ex">-rw-rw-r--</span> 1 vchhatre 66 Oct 28 23:55 CEU.pop.frq</a>
<a class="sourceLine" id="cb11-16" data-line-number="16"><span class="ex">-rw-rw-r--</span> 1 vchhatre 81 Oct 28 23:55 CHB.pop.frq</a>
<a class="sourceLine" id="cb11-17" data-line-number="17"><span class="ex">-rw-rw-r--</span> 1 vchhatre 81 Oct 28 23:55 CHS.pop.frq</a>
<a class="sourceLine" id="cb11-18" data-line-number="18"><span class="ex">-rw-rw-r--</span> 1 vchhatre 80 Oct 28 23:55 CLM.pop.frq</a>
<a class="sourceLine" id="cb11-19" data-line-number="19"><span class="ex">-rw-rw-r--</span> 1 vchhatre 81 Oct 28 23:55 ESN.pop.frq</a>
<a class="sourceLine" id="cb11-20" data-line-number="20"><span class="ex">-rw-rw-r--</span> 1 vchhatre 80 Oct 28 23:55 FIN.pop.frq</a>
<a class="sourceLine" id="cb11-21" data-line-number="21"><span class="ex">-rw-rw-r--</span> 1 vchhatre 66 Oct 28 23:55 GBR.pop.frq</a>
<a class="sourceLine" id="cb11-22" data-line-number="22"><span class="ex">-rw-rw-r--</span> 1 vchhatre 81 Oct 28 23:55 GIH.pop.frq</a>
<a class="sourceLine" id="cb11-23" data-line-number="23"><span class="ex">-rw-rw-r--</span> 1 vchhatre 81 Oct 28 23:55 GWD.pop.frq</a>
<a class="sourceLine" id="cb11-24" data-line-number="24"><span class="ex">-rw-rw-r--</span> 1 vchhatre 66 Oct 28 23:55 IBS.pop.frq</a>
<a class="sourceLine" id="cb11-25" data-line-number="25"><span class="ex">-rw-rw-r--</span> 1 vchhatre 80 Oct 28 23:55 ITU.pop.frq</a>
<a class="sourceLine" id="cb11-26" data-line-number="26"><span class="ex">-rw-rw-r--</span> 1 vchhatre 82 Oct 28 23:55 JPT.pop.frq</a>
<a class="sourceLine" id="cb11-27" data-line-number="27"><span class="ex">-rw-rw-r--</span> 1 vchhatre 82 Oct 28 23:55 KHV.pop.frq</a>
<a class="sourceLine" id="cb11-28" data-line-number="28"><span class="ex">-rw-rw-r--</span> 1 vchhatre 81 Oct 28 23:55 LWK.pop.frq</a>
<a class="sourceLine" id="cb11-29" data-line-number="29"><span class="ex">-rw-rw-r--</span> 1 vchhatre 81 Oct 28 23:55 MSL.pop.frq</a>
<a class="sourceLine" id="cb11-30" data-line-number="30"><span class="ex">-rw-rw-r--</span> 1 vchhatre 80 Oct 28 23:55 MXL.pop.frq</a>
<a class="sourceLine" id="cb11-31" data-line-number="31"><span class="ex">-rw-rw-r--</span> 1 vchhatre 80 Oct 28 23:55 PEL.pop.frq</a>
<a class="sourceLine" id="cb11-32" data-line-number="32"><span class="ex">-rw-rw-r--</span> 1 vchhatre 78 Oct 28 23:55 PJL.pop.frq</a>
<a class="sourceLine" id="cb11-33" data-line-number="33"><span class="ex">-rw-rw-r--</span> 1 vchhatre 80 Oct 28 23:55 PUR.pop.frq</a>
<a class="sourceLine" id="cb11-34" data-line-number="34"><span class="ex">-rw-rw-r--</span> 1 vchhatre 80 Oct 28 23:55 STU.pop.frq</a>
<a class="sourceLine" id="cb11-35" data-line-number="35"><span class="ex">-rw-rw-r--</span> 1 vchhatre 81 Oct 28 23:55 TSI.pop.frq</a>
<a class="sourceLine" id="cb11-36" data-line-number="36"><span class="ex">-rw-rw-r--</span> 1 vchhatre 81 Oct 28 23:55 YRI.pop.frq</a></code></pre></div>
<p>Above, we have truncated the output of the <code>ls</code> command to only include files whose names end in <code>.frq</code>. There are however 52 files in the <code>freq/</code> folder, of which half are <code>.frq</code> files and the other half are <code>.log</code> files from VCFTools. Let’s look inside one of the population allele frequency files.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb12-1" data-line-number="1"></a>
<a class="sourceLine" id="cb12-2" data-line-number="2"><span class="fu">cat</span> LWK.pop.frq </a>
<a class="sourceLine" id="cb12-3" data-line-number="3"></a>
<a class="sourceLine" id="cb12-4" data-line-number="4"><span class="ex">CHROM</span>   POS N_ALLELES   N_CHR   <span class="dt">{ALLELE:FREQ}</span></a>
<a class="sourceLine" id="cb12-5" data-line-number="5"><span class="ex">15</span>  48426484    2   198 A:0.0757576 G:0.924242</a></code></pre></div>
<p>The output is self explanatory. The SNP we are studying (rs1426654) is located on Chromosome 15 at base position 48426484. Being polymorphic, it has two forms (alleles). In population LWK, there are 198 chromosomes altogether (2 homologs per individual @ 99 individuals). Finally, frequencies of both alleles are provided. Notice that the frequencies add up to 1.0 as per expectation.</p>
<p>In order to visualize these allele frequencies for all 26 populations, we first need to combine information from all 26 output files. Simplest way would be to use <code>cat</code> command.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb13-1" data-line-number="1"></a>
<a class="sourceLine" id="cb13-2" data-line-number="2"><span class="fu">cat</span> *.frq <span class="op">&gt;</span> all.freq</a>
<a class="sourceLine" id="cb13-3" data-line-number="3"></a>
<a class="sourceLine" id="cb13-4" data-line-number="4"><span class="fu">head</span> -n 6 all.freq</a>
<a class="sourceLine" id="cb13-5" data-line-number="5"></a>
<a class="sourceLine" id="cb13-6" data-line-number="6"><span class="ex">CHROM</span>   POS N_ALLELES   N_CHR   <span class="dt">{ALLELE:FREQ}</span></a>
<a class="sourceLine" id="cb13-7" data-line-number="7"><span class="ex">15</span>  48426484    2   192 A:0.104167  G:0.895833</a>
<a class="sourceLine" id="cb13-8" data-line-number="8"><span class="ex">CHROM</span>   POS N_ALLELES   N_CHR   <span class="dt">{ALLELE:FREQ}</span></a>
<a class="sourceLine" id="cb13-9" data-line-number="9"><span class="ex">15</span>  48426484    2   122 A:0.188525  G:0.811475</a>
<a class="sourceLine" id="cb13-10" data-line-number="10"><span class="ex">CHROM</span>   POS N_ALLELES   N_CHR   <span class="dt">{ALLELE:FREQ}</span></a>
<a class="sourceLine" id="cb13-11" data-line-number="11"><span class="ex">15</span>  48426484    2   172 A:0.534884  G:0.465116</a>
<a class="sourceLine" id="cb13-12" data-line-number="12"></a></code></pre></div>
<h4 id="formatting-the-all.freq-file">3.2 Formatting the <code>all.freq</code> File</h4>
<p>For our downstream visualization analysis, we need to get this file into correct format. You may have noticed that each allele frequency estimate has a header line. We do not need this redundant information for downstream analysis. Only one instance of the header line should be suffucient. We can selectively extract only those lines that we care about from this file. Let’s grab these lines.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb14-1" data-line-number="1"></a>
<a class="sourceLine" id="cb14-2" data-line-number="2"><span class="fu">sed</span> -n <span class="st">&#39;1p;0~2p&#39;</span> all.freq <span class="op">&gt;</span> master.freq</a>
<a class="sourceLine" id="cb14-3" data-line-number="3"></a>
<a class="sourceLine" id="cb14-4" data-line-number="4"><span class="fu">head</span> master.freq</a>
<a class="sourceLine" id="cb14-5" data-line-number="5"></a>
<a class="sourceLine" id="cb14-6" data-line-number="6"><span class="ex">CHROM</span>   POS N_ALLELES   N_CHR   <span class="dt">{ALLELE:FREQ}</span></a>
<a class="sourceLine" id="cb14-7" data-line-number="7"><span class="ex">15</span>  48426484    2   192 A:0.104167  G:0.895833</a>
<a class="sourceLine" id="cb14-8" data-line-number="8"><span class="ex">15</span>  48426484    2   122 A:0.188525  G:0.811475</a>
<a class="sourceLine" id="cb14-9" data-line-number="9"><span class="ex">15</span>  48426484    2   172 A:0.534884  G:0.465116</a>
<a class="sourceLine" id="cb14-10" data-line-number="10"><span class="ex">15</span>  48426484    2   186 A:0     G:1</a>
<a class="sourceLine" id="cb14-11" data-line-number="11"><span class="ex">15</span>  48426484    2   198 A:1     G:0</a>
<a class="sourceLine" id="cb14-12" data-line-number="12"><span class="ex">15</span>  48426484    2   206 A:0.0291262 G:0.970874</a>
<a class="sourceLine" id="cb14-13" data-line-number="13"><span class="ex">15</span>  48426484    2   210 A:0.0190476 G:0.980952</a>
<a class="sourceLine" id="cb14-14" data-line-number="14"><span class="ex">15</span>  48426484    2   188 A:0.718085  G:0.281915</a>
<a class="sourceLine" id="cb14-15" data-line-number="15"></a></code></pre></div>
<p>The sed command above extracts line 1 (i.e. the first header line), and then every 2nd line from <code>all.freq</code> and outputs them to a new file <code>master.freq</code>. The output contains 27 lines (header + 1 line per population). We do not have population names in this file, but their order is alphabetical. So we should still be able to discern what frequency belongs to which population during the analysis. We need to make two more changes to <code>master.freq</code>. Both require opening it in <code>vim</code>.</p>
<ol type="1">
<li><p>This file has 6 columns, but the column header has only 5 names. Delete the string <code>{ALLELE:FREQ}</code> and replace it with <code>Allele_A</code> <code>&lt;tab&gt;</code> <code>Allele_G</code>. You will need to open the file in vim and then activate <code>INSERT</code> mode to make this change, i.e. press <code>i</code></p></li>
<li><p>Delete the allele names <code>A:</code> and <code>G:</code> as follows. This should be done using the <code>COMMAND</code> mode, i.e. press <code>ESC</code>.</p></li>
</ol>
<div class="sourceCode" id="cb15"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb15-1" data-line-number="1"></a>
<a class="sourceLine" id="cb15-2" data-line-number="2"><span class="ex">vim</span> master.freq</a>
<a class="sourceLine" id="cb15-3" data-line-number="3"></a>
<a class="sourceLine" id="cb15-4" data-line-number="4">[<span class="ex">vim</span>] :%s/A://g</a>
<a class="sourceLine" id="cb15-5" data-line-number="5">[<span class="ex">vim</span>] :%s/G://g</a>
<a class="sourceLine" id="cb15-6" data-line-number="6">[<span class="ex">vim</span>] :wq</a></code></pre></div>
<p>After saving and closing the file make sure it looks as expected:</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb16-1" data-line-number="1"></a>
<a class="sourceLine" id="cb16-2" data-line-number="2">$ <span class="fu">head</span> master.freq</a>
<a class="sourceLine" id="cb16-3" data-line-number="3"></a>
<a class="sourceLine" id="cb16-4" data-line-number="4"><span class="ex">CHROM</span>   POS N_ALLELES   N_CHR   Allele_A    Allele_G</a>
<a class="sourceLine" id="cb16-5" data-line-number="5"><span class="ex">15</span>  48426484    2   192 0.104167    0.895833</a>
<a class="sourceLine" id="cb16-6" data-line-number="6"><span class="ex">15</span>  48426484    2   122 0.188525    0.811475</a>
<a class="sourceLine" id="cb16-7" data-line-number="7"><span class="ex">15</span>  48426484    2   172 0.534884    0.465116</a>
<a class="sourceLine" id="cb16-8" data-line-number="8"><span class="ex">15</span>  48426484    2   186 0       1</a>
<a class="sourceLine" id="cb16-9" data-line-number="9"><span class="ex">15</span>  48426484    2   198 1       0</a>
<a class="sourceLine" id="cb16-10" data-line-number="10"><span class="ex">15</span>  48426484    2   206 0.0291262   0.970874</a>
<a class="sourceLine" id="cb16-11" data-line-number="11"><span class="ex">15</span>  48426484    2   210 0.0190476   0.980952</a>
<a class="sourceLine" id="cb16-12" data-line-number="12"><span class="ex">15</span>  48426484    2   188 0.718085    0.281915</a>
<a class="sourceLine" id="cb16-13" data-line-number="13"><span class="ex">15</span>  48426484    2   198 0.0252525   0.974747</a>
<a class="sourceLine" id="cb16-14" data-line-number="14"></a></code></pre></div>
<h4 id="the-popdata-file">3.3 The <code>popdata</code> File</h4>
<p>At the base of today’s working directory (<code>/project/inbre-train/Week5Data/</code>), there is a file named <code>popdata</code>. It contains alphabetical list of population names whose lines correspond to those in allele frequency file. Check to make sure the file exists. This file also contains information on geographical locations of each of the 26 human populations we are studying. The populations are in turn grouped into 5 super populations.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb17-1" data-line-number="1"><span class="bu">pwd</span></a>
<a class="sourceLine" id="cb17-2" data-line-number="2"></a>
<a class="sourceLine" id="cb17-3" data-line-number="3"><span class="ex">/project/inbre-train</span>/[<span class="ex">inbreNNN</span>]/Week5_LastName/freq</a>
<a class="sourceLine" id="cb17-4" data-line-number="4"></a>
<a class="sourceLine" id="cb17-5" data-line-number="5"><span class="fu">ls</span> -lh ../popdata</a>
<a class="sourceLine" id="cb17-6" data-line-number="6"></a>
<a class="sourceLine" id="cb17-7" data-line-number="7"><span class="ex">-rw-r--r--</span> 1 vchhatre inbre-train 1.6K Oct 28 22:23 popdata</a>
<a class="sourceLine" id="cb17-8" data-line-number="8"></a></code></pre></div>
<h4 id="join-popdata-with-master.freq">3.4 Join <code>popdata</code> with <code>master.freq</code></h4>
<p>The simplest way to accomplish this is using <code>paste</code> utility.</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb18-1" data-line-number="1"></a>
<a class="sourceLine" id="cb18-2" data-line-number="2"><span class="ex">paste</span> ../popdata master.freq <span class="op">&gt;</span> freq.df</a>
<a class="sourceLine" id="cb18-3" data-line-number="3"></a>
<a class="sourceLine" id="cb18-4" data-line-number="4"><span class="fu">cat</span> freq.df</a>
<a class="sourceLine" id="cb18-5" data-line-number="5"></a>
<a class="sourceLine" id="cb18-6" data-line-number="6"><span class="ex">pop</span>     dist    superpop        lat     long    popname CHROM   POS     N_ALLELES       N_CHR   Allele_A        Allele_G</a>
<a class="sourceLine" id="cb18-7" data-line-number="7"><span class="ex">ACB</span>     13.19   AFR     13.1776 -59.5412        African_Carib_BBDS      15      48426484        2       192     0.104167        0.895833</a>
<a class="sourceLine" id="cb18-8" data-line-number="8"><span class="ex">ASW</span>     -8.78   AFR     36.1070 -112.1130       African_Ancestry_SW_USA 15      48426484        2       122     0.188525        0.811475</a>
<a class="sourceLine" id="cb18-9" data-line-number="9"><span class="ex">BEB</span>     23.68   SAS     23.6850 90.3563 Bengali_in_Bangladesh   15      48426484        2       172     0.534884        0.465116</a>
<a class="sourceLine" id="cb18-10" data-line-number="10"><span class="ex">CDX</span>     22.01   EAS     22.0088 -100.7971       Chinese_Dai     15      48426484        2       186     0       1</a>
<a class="sourceLine" id="cb18-11" data-line-number="11"><span class="ex">CEU</span>     62.28   EUR     39.3210 -111.0937       Utah_Resid_from_NWEurope        15      48426484        2       198     1       0</a>
<a class="sourceLine" id="cb18-12" data-line-number="12"><span class="ex">CHB</span>     23.13   EAS     39.9042 116.4074        Han_Chinese     15      48426484        2       206     0.0291262       0.970874</a>
<a class="sourceLine" id="cb18-13" data-line-number="13"><span class="ex">CHS</span>     24.48   EAS     23.9790 113.7633        Southern_Han_Chinese    15      48426484        2       210     0.0190476       0.980952</a>
<a class="sourceLine" id="cb18-14" data-line-number="14"><span class="ex">CLM</span>     6.24    AMR     6.2442  -75.5812        Columbian_Medellin      15      48426484        2       188     0.718085        0.281915</a>
<a class="sourceLine" id="cb18-15" data-line-number="15"><span class="ex">ESN</span>     10.22   AFR     9.0820  8.6753  Esan_in_Nigeria 15      48426484        2       198     0.0252525       0.974747</a>
<a class="sourceLine" id="cb18-16" data-line-number="16"><span class="ex">FIN</span>     61.92   EUR     61.9241 25.7482 Finnish_Finland 15      48426484        2       198     0.989899        0.010101</a>
<a class="sourceLine" id="cb18-17" data-line-number="17"><span class="ex">GBR</span>     56.49   EUR     55.3781 -3.4360 Bri_England_Scotland    15      48426484        2       182     1       0</a>
<a class="sourceLine" id="cb18-18" data-line-number="18"><span class="ex">GIH</span>     22.26   SAS     22.2587 71.1924 Guj_Ind_Houston 15      48426484        2       206     0.951456        0.0485437</a>
<a class="sourceLine" id="cb18-19" data-line-number="19"><span class="ex">GWD</span>     13.44   AFR     13.4432 -15.3101        Gambian_WestGambia      15      48426484        2       226     0.0752212       0.924779</a>
<a class="sourceLine" id="cb18-20" data-line-number="20"><span class="ex">IBS</span>     40.46   EUR     40.4830 -4.0876 Iberian_Spain   15      48426484        2       214     1       0</a>
<a class="sourceLine" id="cb18-21" data-line-number="21"><span class="ex">ITU</span>     11.13   SAS     11.1271 78.6569 Telugu_Ind_UK   15      48426484        2       204     0.651961        0.348039</a>
<a class="sourceLine" id="cb18-22" data-line-number="22"><span class="ex">JPT</span>     35.69   EAS     35.6895 139.6917        Jap_Tokyo       15      48426484        2       208     0.00480769      0.995192</a>
<a class="sourceLine" id="cb18-23" data-line-number="23"><span class="ex">KHV</span>     14.06   EAS     14.0583 108.2772        Kinh_Vietnam    15      48426484        2       198     0.00505051      0.994949</a>
<a class="sourceLine" id="cb18-24" data-line-number="24"><span class="ex">LWK</span>     -0.02   AFR     -0.0236 37.9062 Luhya_Kenya     15      48426484        2       198     0.0757576       0.924242</a>
<a class="sourceLine" id="cb18-25" data-line-number="25"><span class="ex">MSL</span>     8.46    AFR     8.4606  -11.7799        Mende_Sierra_Leone      15      48426484        2       170     0.0882353       0.911765</a>
<a class="sourceLine" id="cb18-26" data-line-number="26"><span class="ex">MXL</span>     23.63   AMR     23.6345 -102.5528       Mex_LA_USA      15      48426484        2       128     0.515625        0.484375</a>
<a class="sourceLine" id="cb18-27" data-line-number="27"><span class="ex">PEL</span>     -9.19   AMR     -9.19   -75.0152        Peruvian        15      48426484        2       170     0.282353        0.717647</a>
<a class="sourceLine" id="cb18-28" data-line-number="28"><span class="ex">PJL</span>     31.55   SAS     31.5546 74.3572 Punjabi_Lahore  15      48426484        2       192     0.78125 0.21875</a>
<a class="sourceLine" id="cb18-29" data-line-number="29"><span class="ex">PUR</span>     18.22   AMR     18.2208 -66.5901        Puerto_Rican    15      48426484        2       208     0.769231        0.230769</a>
<a class="sourceLine" id="cb18-30" data-line-number="30"><span class="ex">STU</span>     7.87    SAS     7.8731  80.7718 Sri_Lankan_Tamil        15      48426484        2       204     0.485294        0.514706</a>
<a class="sourceLine" id="cb18-31" data-line-number="31"><span class="ex">TSI</span>     43.77   EUR     43.7711 11.2486 Toscani_Italy   15      48426484        2       214     0.995327        0.0046729</a>
<a class="sourceLine" id="cb18-32" data-line-number="32"><span class="ex">YRI</span>     10.16   AFR     7.3775  3.9470  Yoruba_Ibadan_Nigeria   15      48426484        2       216     0.0138889       0.986111</a></code></pre></div>
<h4 id="download-the-freq.df-file-to-your-local-workstation">3.5 Download the <code>freq.df</code> File to Your Local Workstation</h4>
<p>Recall that in order to copy files from server to local workstation, you will first need to open a new terminal window (<code>Cmd+Tab</code>).</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb19-1" data-line-number="1"></a>
<a class="sourceLine" id="cb19-2" data-line-number="2"><span class="bu">cd</span> ~/</a>
<a class="sourceLine" id="cb19-3" data-line-number="3"></a>
<a class="sourceLine" id="cb19-4" data-line-number="4"><span class="fu">mkdir</span> LASTNAME_week5</a>
<a class="sourceLine" id="cb19-5" data-line-number="5"></a>
<a class="sourceLine" id="cb19-6" data-line-number="6"><span class="bu">cd</span> LASTNAME_week5/</a>
<a class="sourceLine" id="cb19-7" data-line-number="7"></a>
<a class="sourceLine" id="cb19-8" data-line-number="8"><span class="fu">scp</span> [inbreNNN]@teton.uwyo.edu:/project/inbre-train/<span class="op">&lt;</span>inbreNNN<span class="op">&gt;</span>/Week5_LastName/freq/freq.df .</a>
<a class="sourceLine" id="cb19-9" data-line-number="9"></a></code></pre></div>
</body>
</html>
