<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
  <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
  <meta http-equiv="Content-Style-Type" content="text/css" />
  <meta name="generator" content="pandoc" />
  <title></title>
  <style type="text/css">code{white-space: pre;}</style>
  <style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
  </style>
  <link rel="stylesheet" href="avenir-white.css" type="text/css" />
</head>
<body>
<h1 id="human-population-genomic-variation">Human Population Genomic Variation</h1>
<h2 id="variants-snps-in-lct-gene-part-1">Variants (SNPs) in <em>LCT</em> Gene: Part 1</h2>
<h3 id="molb-44855485----computers-in-biology">Molb 4485/5485 -- Computers in Biology</h3>
<h3 id="vikram-chhatre-and-nicolas-blouin">Vikram Chhatre and Nicolas Blouin</h3>
<p>Wyoming INBRE Bioinformatics Core<br />
Dept. of Molecular Biology<br />
University of Wyoming</p>
<a href="mailto:vchhatre@uwyo.edu">vchhatre@uwyo.edu</a><br />
<a href="mailto:nblouin@uwyo.edu">nblouin@uwyo.edu</a><br />
<a href="http://molb4485.uwyo.online" class="uri">http://molb4485.uwyo.online</a><br />

<hr>
<p><img src="hg.jpg" width=800></a> Hunting and Gathering to Pastoralism. Image Credit: <a href="http://dburr.blogspot.com">Dan Burr</a></p>
<h3 style="color: red;">
Important:
</h3>
<p><strong>1. Do not race ahead in the tutorial.</strong></p>
<p><strong>2. Doing so compromises your learning experience.</strong></p>
<p><strong>3. You may not leave early today!</strong> <br><br><br></p>
<h2 id="table-of-contents">Table of Contents</h2>
<ol style="list-style-type: decimal">
<li><a href="#data">Exercise Data</a></li>
<li><a href="#vcf">Variant Call Format</a></li>
<li><a href="#freq">Gene Frequency</a></li>
</ol>
<p><br><br><br><br><br></p>
<p><a name="data"></a></p>
<h3 id="data-for-todays-exercise">1. Data For Today's Exercise</h3>
<p>For this entire exercise, replace <code>[inbreNNN]</code> with your own <code>inbre</code> ID. Once you log on to Mt. Moran, copy the data for today's exercise as follows.</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash">$ <span class="fu">ssh</span> [inbreNNN]@mtmoran.uwyo.edu

$ <span class="bu">cd</span> /project/inbre-train/[inbreNNN]/

$ <span class="fu">mkdir</span> Week5 <span class="kw">&amp;&amp;</span> <span class="bu">cd</span> Week5/

$ <span class="fu">cp</span> -r /project/inbre-train/Week5Data/* .

$ <span class="fu">ls</span> -l
</code></pre></div>
<p><a name="vcf"></a></p>
<h3 id="variant-call-format-vcf-spec.-4.1">2. Variant Call Format (VCF) spec. 4.1</h3>
<p>This is a standard for storing information on polymorphism in genes. It is a <code>n x n</code> matrix of individuals and variants. Columns refer to individuals, whereas rows refer to variants (SNPs). Analogous to the FASTQ format, VCF also stores quality information. Files formatted to store variant calls always end in <code>.vcf</code> extension. For example: <code>input.vcf</code>. The VCF format is described in detail in the user manual available at <a href="http://samtools.github.io" class="uri">http://samtools.github.io</a>.</p>
<h4 id="vcftools">2.1 VCFTools</h4>
<p>VCFTools is a program that can report information on and manipulate the .vcf files. A compre- hensive manual on VCFTools usage is <a href="http://vcftools.sourceforge.net/man%20latest.html">available here</a>. Before we can use this program, we must load it in memory as a module.</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash">
$ <span class="ex">module</span> spider vcftools

$ <span class="ex">module</span> load intel perl vcftools

$ <span class="ex">vcftools</span>

<span class="ex">VCFtools</span> (0.1.15)
    <span class="ex">Adam</span> Auton and Anthony Marcketta 2009
</code></pre></div>
<h4 id="vcf-file">2.2 VCF File</h4>
<p>We have provided you with an example <code>.vcf</code> file for today’s exercise: <code>human_chr2.vcf</code>. As the name suggests, this file contains data from <em>Homo sapiens</em> <code>CHROMOSOME 2</code>. The data was collected as part of the <a href="http://www.internationalgenome.org">1000 Genomes Project</a> whose aim is to understand human genomic variation across the globe. This data is from the final phase of the project and contains genotypes of ~2500 individuals from 26 populations distributed globally genotyped at more than 84 million SNPs.</p>
<p>To make it more manageable, we have subset the original data to only include information on 1000 SNPs. You will be further subsetting this file to obtain data on one specific SNP that we are interested in. But for now, open the file in text editor and take a look at the contents. Then close the file without saving.</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash">
$ <span class="ex">vim</span> human_chr2.vcf

[<span class="ex">vim</span>] :set nowrap
[<span class="ex">vim</span>] :wq</code></pre></div>
<h4 id="vcf-data-summary">2.3 VCF Data Summary</h4>
<p>VCFTools can provide a quick summary of your data. When running without any options, VCFTools outputs summary statistics but does not make any changes to the file nor does it produce any output files.</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash">
$ <span class="ex">vcftools</span> --vcf human_chr2.vcf

<span class="ex">VCFtools</span> - 0.1.15
   <span class="kw">(</span><span class="ex">C</span><span class="kw">)</span> <span class="ex">Adam</span> Auton and Anthony Marcketta 2009

   <span class="ex">Parameters</span> as interpreted:
           <span class="ex">--vcf</span> human_chr2.vcf

   <span class="ex">After</span> filtering, kept 2504 out of 2504 Individuals
   <span class="ex">After</span> filtering, kept 1000 out of a possible 1000 Sites

   <span class="ex">Run</span> Time = 0.00 seconds</code></pre></div>
<h4 id="snp-rs34100645">2.4 SNP <code>rs34100645</code></h4>
<p>For today's exercise, we are interested in only one SNP: <code>rs776746</code>. Every documented SNP from the human genome has a <code>rs</code> ID associated with it. Note that the number following rs, is not the physical position of this SNP on a chromosome, but simply an identifier. You can look up the physical position of this SNP in the VCF file. For example:</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash">
$ <span class="fu">grep</span> <span class="st">&quot;rs34100645&quot;</span> human_chr2.vcf <span class="kw">|</span> <span class="fu">cut</span> -f1-5

<span class="ex">2</span>   135805941   rs34100645  A   T</code></pre></div>
<p>Why did we pass the output of grep to cut via a <code>|</code>? It's because when grep matches a pattern, it prints the entire line to the screen. Our lines here have more than 2500 columns and we are only interested in the first few columns as displayed above.</p>
<p>This is one of more than 1000 SNPs present within the sequence of gene <em>LCT</em> or Lactose Persistence. One of the alleles at this mutation (SNP) has a very small contribution to the ability of digesting lactose sugar present in the milk. Thus, we are interested in the frequency of this allele in various human populations.</p>
<h4 id="subset-vcf-for-rs34100645">2.5 Subset VCF for <code>rs34100645</code></h4>
<p>While there are more than 1000 SNPs in our vcf file, we only need one for today's exercise. So let's go ahead and subset the VCF for that SNP.</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash">$ <span class="ex">vcftools</span> --vcf human_chr2.vcf --snp rs34100645 --recode --out lactose


<span class="ex">VCFtools</span> - 0.1.15
<span class="kw">(</span><span class="ex">C</span><span class="kw">)</span> <span class="ex">Adam</span> Auton and Anthony Marcketta 2009

<span class="ex">Parameters</span> as interpreted:
    <span class="ex">--vcf</span> human_chr2.vcf
    <span class="ex">--out</span> lactose
    <span class="ex">--recode</span>
    <span class="ex">--snp</span> rs34100645

<span class="ex">After</span> filtering, kept 2504 out of 2504 Individuals
<span class="ex">Outputting</span> VCF file...
<span class="ex">After</span> filtering, kept 1 out of a possible 1248 Sites
<span class="ex">Run</span> Time = 0.00 seconds</code></pre></div>
<p>In the above command, we extracted data for SNP rs34100645 from the file human_chr2.vcf and provided a prefix <code>lactose</code> for the output file. Check the output:</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash">
$ <span class="fu">ls</span> -l

<span class="ex">-rw-rw-r--</span> 1 [inbreNNN] inbre-train 12572407 Nov 12 16:19 human_chr2.vcf
<span class="ex">-rw-rw-r--</span> 1 [inbreNNN] inbre-train      306 Nov 12 16:21 lactose.log
<span class="ex">-rw-rw-r--</span> 1 [inbreNNN] inbre-train    30135 Nov 12 16:21 lactose.recode.vcf</code></pre></div>
<p>The information that was printed to the screen was also populated inside the log file <code>lactose.log</code>. Your main output file is <code>lactose.recode.vcf</code>, but let's change its name to <code>lactose.vcf</code> for convenience.</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash">

$ <span class="fu">mv</span> lactose.recode.vcf lactose.vcf

$ <span class="fu">ls</span> -l

<span class="ex">-rw-rw-r--</span> 1 [inbreNNN] inbre-train    30135 Nov 12 16:21 lactose.vcf</code></pre></div>
<p><a name="freq"></a></p>
<h2 id="gene-frequency">3. Gene Frequency</h2>
<p>As we discussed earlier, allele frequency is a measure of the relative proportion of the two forms of a gene in a population. If a given SNP is not polymorphic, it will have only one form. Frequency of that form will be <code>f = 1.00</code> or 100%. However, if it is polymorphic, the resulting gene forms (alleles) may either be present at equal proportion (<code>f(p) = f(q) = 0.5</code>) or at variable proportions such as <code>f(p) = 0.25</code> and <code>f(q) = 0.75</code>. No matter their relative proportion, the frequencies of the two alleles will always add up to 1 in any population sample.</p>
<h3 id="frequency-of-rs34100645-alleles">3.1 Frequency of <code>rs34100645</code> Alleles</h3>
<p>We can use VCFTools to quickly estimate frequencies of the two alleles (<code>A/T</code>). But remember that allele frequency is a measure of population(s). During today's exercise, we want to calculate these frequencies in each of the 26 human populations that we have data for. However, the vcf files does not contain any information on which individuals belong to which populations. Instead the population classification information is present in the folder <code>popfiles</code>:</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash">
$ <span class="fu">ls</span> -l popfiles/
<span class="ex">total</span> 0

<span class="ex">-rw-rw-r--</span> 1 [inbreNNN] inbre-train 768 Nov 12 16:31 ACB.pop
<span class="ex">-rw-rw-r--</span> 1 [inbreNNN] inbre-train 488 Nov 12 16:31 ASW.pop
<span class="ex">-rw-rw-r--</span> 1 [inbreNNN] inbre-train 688 Nov 12 16:31 BEB.pop
<span class="ex">-rw-rw-r--</span> 1 [inbreNNN] inbre-train 744 Nov 12 16:31 CDX.pop
<span class="ex">-rw-rw-r--</span> 1 [inbreNNN] inbre-train 792 Nov 12 16:31 CEU.pop
<span class="ex">-rw-rw-r--</span> 1 [inbreNNN] inbre-train 824 Nov 12 16:31 CHB.pop
<span class="ex">-rw-rw-r--</span> 1 [inbreNNN] inbre-train 840 Nov 12 16:31 CHS.pop
<span class="ex">-rw-rw-r--</span> 1 [inbreNNN] inbre-train 752 Nov 12 16:31 CLM.pop
<span class="ex">-rw-rw-r--</span> 1 [inbreNNN] inbre-train 792 Nov 12 16:31 ESN.pop
<span class="ex">-rw-rw-r--</span> 1 [inbreNNN] inbre-train 792 Nov 12 16:31 FIN.pop
<span class="ex">-rw-rw-r--</span> 1 [inbreNNN] inbre-train 728 Nov 12 16:31 GBR.pop
<span class="ex">-rw-rw-r--</span> 1 [inbreNNN] inbre-train 824 Nov 12 16:31 GIH.pop
<span class="ex">-rw-rw-r--</span> 1 [inbreNNN] inbre-train 904 Nov 12 16:31 GWD.pop
<span class="ex">-rw-rw-r--</span> 1 [inbreNNN] inbre-train 856 Nov 12 16:31 IBS.pop
<span class="ex">-rw-rw-r--</span> 1 [inbreNNN] inbre-train 816 Nov 12 16:31 ITU.pop
<span class="ex">-rw-rw-r--</span> 1 [inbreNNN] inbre-train 832 Nov 12 16:31 JPT.pop
<span class="ex">-rw-rw-r--</span> 1 [inbreNNN] inbre-train 792 Nov 12 16:31 KHV.pop
<span class="ex">-rw-rw-r--</span> 1 [inbreNNN] inbre-train 792 Nov 12 16:31 LWK.pop
<span class="ex">-rw-rw-r--</span> 1 [inbreNNN] inbre-train 680 Nov 12 16:31 MSL.pop
<span class="ex">-rw-rw-r--</span> 1 [inbreNNN] inbre-train 512 Nov 12 16:31 MXL.pop
<span class="ex">-rw-rw-r--</span> 1 [inbreNNN] inbre-train 680 Nov 12 16:31 PEL.pop
<span class="ex">-rw-rw-r--</span> 1 [inbreNNN] inbre-train 768 Nov 12 16:31 PJL.pop
<span class="ex">-rw-rw-r--</span> 1 [inbreNNN] inbre-train 832 Nov 12 16:31 PUR.pop
<span class="ex">-rw-rw-r--</span> 1 [inbreNNN] inbre-train 816 Nov 12 16:31 STU.pop
<span class="ex">-rw-rw-r--</span> 1 [inbreNNN] inbre-train 856 Nov 12 16:31 TSI.pop
<span class="ex">-rw-rw-r--</span> 1 [inbreNNN] inbre-train 864 Nov 12 16:31 YRI.pop
</code></pre></div>
<p>Each of these popfiles contains a list of individuals that belong to that specific population. Try using the linux command <code>cat</code> on some of the files to see what is inside. Next, we will be writing a simple shell script to automate the estimation of allele frequencies using these population files and VCFTools. This script consumes very little computational resources, so we will not run it using sbatch. Thus you can exclude the scheduler commands that begin with #SBATCH for this script.</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash">

$ <span class="ex">vim</span> makefreq.sh

   <span class="co">#!/bin/bash</span>

   <span class="co">## Make folder to store frequency output</span>
   <span class="fu">mkdir</span> freq

   <span class="co">## Navigate to the folder ‘popfiles/’</span>
   <span class="bu">cd</span> popfiles/

   <span class="co">## For loop</span>
   <span class="kw">for</span> <span class="ex">p</span> in *.pop
   <span class="kw">do</span>
     <span class="ex">vcftools</span> --vcf ../lactose.vcf --keep <span class="va">$p</span> --freq --out ../freq/<span class="va">$p</span>
   <span class="kw">done</span>
   <span class="bu">cd</span> ..
</code></pre></div>
<p>Save and close the file. This script demonstrates a simple for loop which iterates over popfiles shown above and executes vcftools (to estimate allele frequencies) for only the individuals in the population file currently under iteration. Go ahead and execute it. Then look at the results.</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash">
$ <span class="fu">bash</span> makefreq.sh

$ <span class="bu">pwd</span>
<span class="ex">/project/inbre-train</span>/[<span class="ex">inbreNNN</span>]/LASTNAME_Week5

$ <span class="bu">cd</span> freq

$ <span class="fu">ls</span> -l

<span class="ex">-rw-rw-r--</span> 1 [inbreNNN] inbre-train 37111 Nov 12 16:48 ACB.pop.frq
<span class="ex">-rw-rw-r--</span> 1 [inbreNNN] inbre-train 50000 Nov 12 16:48 ACB.pop.log

<span class="ex">-rw-rw-r--</span> 1 [inbreNNN] inbre-train 36716 Nov 12 16:48 ASW.pop.frq
<span class="ex">-rw-rw-r--</span> 1 [inbreNNN] inbre-train 50000 Nov 12 16:48 ASW.pop.log

<span class="ex">-rw-rw-r--</span> 1 [inbreNNN] inbre-train 36201 Nov 12 16:48 BEB.pop.frq
<span class="ex">-rw-rw-r--</span> 1 [inbreNNN] inbre-train 50000 Nov 12 16:48 BEB.pop.log

<span class="ex">-rw-rw-r--</span> 1 [inbreNNN] inbre-train 35319 Nov 12 16:48 CDX.pop.frq
<span class="ex">-rw-rw-r--</span> 1 [inbreNNN] inbre-train 50000 Nov 12 16:48 CDX.pop.log

<span class="ex">-rw-rw-r--</span> 1 [inbreNNN] inbre-train 35082 Nov 12 16:48 CEU.pop.frq
<span class="ex">-rw-rw-r--</span> 1 [inbreNNN] inbre-train 50000 Nov 12 16:48 CEU.pop.log
</code></pre></div>
<p>Above, we have truncated the output of the ls command after the first four lines. You should be seeing 52 files altogether, of which half are freq files and the other half are log files from VCFTools. Let's look inside one of the population allele frequency files.</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash">
$ <span class="fu">cat</span> ACB.pop.frq 

<span class="ex">CHROM</span>   POS     N_ALLELES   N_CHR   <span class="dt">{ALLELE:FREQ}</span>
<span class="ex">2</span>   135805941       2   192 A:0.6875    T:0.3125</code></pre></div>
<p>The output is self explanatory. The SNP we are studying (rs34100645) is located on Chromosome 2 at base position 135805941. Being polymorphic, it has two forms (alleles). In population ACB, there are 192 chromosomes altogether (2 per individual @ 96 individuals). Finally, frequencies of both alleles are provided. Notice that the frequencies add up to 1.0 as per expectation.</p>
<p>In order to visualize these allele frequencies for all 26 populations, we first need to combine information from all 26 output files. Simplest way would be to use <code>cat</code> command.</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash">
$ <span class="fu">cat</span> *.frq <span class="op">&gt;</span> all.freq

$ <span class="fu">head</span> -n 6 all.freq

<span class="ex">CHROM</span>   POS N_ALLELES   N_CHR   <span class="dt">{ALLELE:FREQ}</span>
<span class="ex">2</span>   135805941   2   192 A:0.6875    T:0.3125
<span class="ex">CHROM</span>   POS N_ALLELES   N_CHR   <span class="dt">{ALLELE:FREQ}</span>
<span class="ex">2</span>   135805941   2   122 A:0.688525  T:0.311475
<span class="ex">CHROM</span>   POS N_ALLELES   N_CHR   <span class="dt">{ALLELE:FREQ}</span>
<span class="ex">2</span>   135805941   2   172 A:0.639535  T:0.360465
</code></pre></div>
<h4 id="formatting-the-all.freq-file">3.2 Formatting the <code>all.freq</code> File</h4>
<p>For our downstream visualization analysis, we need to get this file into correct format. You may have noticed that each allele frequency estimate has a header line. We do not need this redundant information for downstream analysis. Only one instance of the header line should be su cient. We can selectively extract only those lines that we care about from this file. Let's grab these lines.</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash">
$ <span class="fu">sed</span> -n <span class="st">&#39;1p;0~2p&#39;</span> all.freq <span class="op">&gt;</span> master.freq

$ <span class="fu">head</span> master.freq

<span class="ex">CHROM</span>   POS N_ALLELES   N_CHR   <span class="dt">{ALLELE:FREQ}</span>
<span class="ex">2</span>   135805941   2   192 A:0.6875    T:0.3125
<span class="ex">2</span>   135805941   2   122 A:0.688525  T:0.311475
<span class="ex">2</span>   135805941   2   172 A:0.639535  T:0.360465
<span class="ex">2</span>   135805941   2   186 A:0.580645  T:0.419355
<span class="ex">2</span>   135805941   2   198 A:0.914141  T:0.0858586
<span class="ex">2</span>   135805941   2   206 A:0.762136  T:0.237864
<span class="ex">2</span>   135805941   2   210 A:0.719048  T:0.280952
<span class="ex">2</span>   135805941   2   188 A:0.765957  T:0.234043
<span class="ex">2</span>   135805941   2   198 A:0.631313  T:0.368687
</code></pre></div>
<p>The sed command above extracts line 1 (i.e. the first header line), and then every 2nd line from <code>all.freq</code> and outputs them to a new file <code>master.freq</code>. The output contains 27 lines (header + 1 line per population). We do not have population names in this file, but their order is alphabetical. So we should still be able to discern what frequency belongs to which population during the analysis. We need to make two more changes to <code>master.freq</code>. Both require opening it in <code>vim</code>.</p>
<ol style="list-style-type: decimal">
<li><p>This file has 6 columns, but the column header has only 5 names. Delete the string <code>{ALLELE:FREQ}</code> and replace it with <code>Allele_A</code> <code>TAB</code> <code>Allele_T</code>. You will need to open the file in vim and then activate <code>INSERT</code> mode to make this change, i.e. press <code>i</code></p></li>
<li><p>Delete the allele names <code>A:</code> and <code>T:</code> as follows. This should be done using the <code>COMMAND</code> mode, i.e. press <code>ESC</code>.</p></li>
</ol>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash">
$ <span class="ex">vim</span> master.freq

:<span class="ex">%s</span>/A:<span class="ex">//g</span>
:<span class="ex">%s</span>/T:<span class="ex">//g</span>
:<span class="ex">wq</span></code></pre></div>
<p>After saving and closing the file make sure it looks as expected:</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash">
$ <span class="fu">head</span> master.freq

<span class="ex">CHROM</span>   POS N_ALLELES   N_CHR   Allele_A    Allele_T
<span class="ex">2</span>   135805941   2   192 0.6875      0.3125
<span class="ex">2</span>   135805941   2   122 0.688525    0.311475
<span class="ex">2</span>   135805941   2   172 0.639535    0.360465
<span class="ex">2</span>   135805941   2   186 0.580645    0.419355
<span class="ex">2</span>   135805941   2   198 0.914141    0.0858586
<span class="ex">2</span>   135805941   2   206 0.762136    0.237864
<span class="ex">2</span>   135805941   2   210 0.719048    0.280952
<span class="ex">2</span>   135805941   2   188 0.765957    0.234043
<span class="ex">2</span>   135805941   2   198 0.631313    0.368687</code></pre></div>
<h4 id="the-popdata-file">3.3 The <code>popdata</code> File</h4>
<p>At the base of today's working directory (<code>/project/inbre-train/Week5Data/</code>), there is a file named <code>popdata</code>. It contains alphabetical list of population names whose lines correspond to those in allele frequency file. Check to make sure the file exists. This file also contains information on geographical locations of each of the 26 human populations we are studying. The populations are in turn grouped into 5 super populations.</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash">$ <span class="bu">pwd</span>

<span class="ex">/project/inbre-train</span>/[<span class="ex">inbreNNN</span>]/Week5/freq

$ <span class="fu">ls</span> -l ../podata

<span class="ex">-rw-rw-r--</span> 1 [inbreNNN] inbre-train 1089 Nov 12 17:24 popdata
</code></pre></div>
<h4 id="join-popdata-with-master.freq">3.4 Join <code>popdata</code> with <code>master.freq</code></h4>
<p>The simplest way to accomplish this is using <code>paste</code> utility.</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash">
$ <span class="ex">paste</span> ../popdata master.freq <span class="op">&gt;</span> freq.df

$ <span class="fu">head</span> freq.df

<span class="ex">pop</span>     dist    superpop        lat     long    popname CHROM   POS N_ALLELES   N_CHR   Allele_A    Allele_T
<span class="ex">ACB</span>     13.19   AFR     13.1776 -59.5412        African_Carib_BBDS  2   135805941   2   192 0.6875  0.3125
<span class="ex">ASW</span>     -8.78   AFR     36.1070 -112.1130       African_Ancestry_SW_USA 2   135805941   2   122 0.688525    0.311475
<span class="ex">BEB</span>     23.68   SAS     23.6850 90.3563 Bengali_in_Bangladesh   2   135805941   2   172 0.639535    0.360465
<span class="ex">CDX</span>     22.01   EAS     22.0088 -100.7971       Chinese_Dai 2   135805941   2   186 0.580645    0.419355
<span class="ex">CEU</span>     62.28   EUR     39.3210 -111.0937       Utah_Resid_from_NWEurope    2   135805941   2   198 0.914141    0.0858586
<span class="ex">CHB</span>     23.13   EAS     39.9042 116.4074        Han_Chinese 2   135805941   2   206 0.762136    0.237864
<span class="ex">CHS</span>     24.48   EAS     23.9790 113.7633        Southern_Han_Chinese    2   135805941   2   210 0.719048    0.280952
<span class="ex">CLM</span>     6.24    AMR     6.2442  -75.5812        Columbian_Medellin  2   135805941   2   188 0.765957    0.234043
<span class="ex">ESN</span>     10.22   AFR     9.0820  8.6753  Esan_in_Nigeria 2   135805941   2   198 0.631313    0.368687
</code></pre></div>
<h4 id="download-the-freq.df-file-to-workstation">3.5 Download the <code>freq.df</code> File to Workstation</h4>
<p>Recall that in order to copy files from server to local workstation, you will first need to open a new terminal window (<code>Cmd+Tab</code>).</p>
<div class="sourceCode"><pre class="sourceCode bash"><code class="sourceCode bash">
$ <span class="bu">cd</span> ~/

$ <span class="fu">mkdir</span> LASTNAME_week5

$ <span class="bu">cd</span> LASTNAME_week5/

$ <span class="fu">scp</span> [inbreNNN]@mtmoran.uwyo.edu:/project/inbre-train/[inbreNNN]/Week5/freq.df .
</code></pre></div>
</body>
</html>
