<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">
<head>
  <meta charset="utf-8" />
  <meta name="generator" content="pandoc" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
  <title>Molb 4485/5485 – Computers in Biology – Syllabus Fall Semester</title>
  <style>
      code{white-space: pre-wrap;}
      span.smallcaps{font-variant: small-caps;}
      span.underline{text-decoration: underline;}
      div.column{display: inline-block; vertical-align: top; width: 50%;}
  </style>
  <link rel="stylesheet" href="avenir-white.css" />
  <!--[if lt IE 9]>
    <script src="//cdnjs.cloudflare.com/ajax/libs/html5shiv/3.7.3/html5shiv-printshiv.min.js"></script>
  <![endif]-->
</head>
<body>
<header id="title-block-header">
<h1 class="title">Molb 4485/5485 – Computers in Biology – Syllabus Fall Semester</h1>
</header>
<h3 id="vikram-e.-chhatre-phd-instructor-2">Vikram E. Chhatre, PhD &amp; Instructor 2</h3>
<p>Wyoming INBRE Bioinformatics Core<br />
Dept. of Molecular Biology<br />
University of Wyoming</p>
<p><strong>Email:</strong> <a href="mailto:vchhatre@uwyo.edu">vchhatre@uwyo.edu</a></p>
<hr>
<h3 id="course-description">1. Course Description</h3>
<p>Technological advances have led to exponential increase in the size of biological data sets. Handling of these large datasets has become intractable without some level of bioinformatic literacy. Many biologists find that there is a steep learning curve to develop the skills required to explore their datasets effectively because of their size or complexity. This bioinformatics short course includes a rich collection of hands-on instruction and lectures specifically intended to help novice users become comfortable with a range of tools currently used to analyze such data. All coursework will be implemented in linux on local workstations or a remote server.</p>
<h4 id="prerequisites">1.1 Prerequisites</h4>
<p>There is no prerequisite for this course other than a willingness to learn and to work hard throughout the 5 weeks.</p>
<h3 id="student-learning-outcomes">2. Student Learning Outcomes</h3>
<p><strong>(also represents expectations for student performance)</strong></p>
<ul>
<li><p>Be familiar with command line interface and working remotely on a high performance computer (i.e., super computer).</p></li>
<li><p>Comfortable making, editing, and moving files.</p></li>
<li><p>Know how to use basic Linux and R commands.</p></li>
<li><p>Write simple scripts to schedule and perform tasks remotely.</p></li>
<li><p>Perform and interpret sequence homology searches, sequence alignments, and phylogenetic reconstruction.</p></li>
<li><p>Familiarity with handling genomic data generated by NextGen platforms including DNA/RNA Sequence quality assessment and data preparation.</p></li>
</ul>
<h3 id="course-structure">3. Course Structure:</h3>
<p>The course is organized into 5 modules</p>
<ul>
<li><p><strong>Module 1:</strong> Course introduction, linux command line interface &amp; text editors</p></li>
<li><p><strong>Module 2:</strong> Working on a computer cluster, sequence similarity searches</p></li>
<li><p><strong>Module 3:</strong> Sequence alignments and phylogenetics<br />
</p></li>
<li><p><strong>Module 4:</strong> Next-generation sequencing and quality assessment</p></li>
<li><p><strong>Module 5:</strong> Variant discovery and analysis across global human populations</p></li>
</ul>
<h3 id="texts-readings-and-access-to-materials">4. Text(s), Readings, and Access to Materials:</h3>
<p>All course materials will be disseminated during (or before) class meetings. These materials will also be available after each class meeting through WyoCourses. Additional reference or working materials will be suggested throughout the course for those interested in exploring topics in a deeper manner.</p>
<h3 id="assignments-and-grading-standards">5. Assignments and Grading Standards:</h3>
<p>All grading will be done on an individual basis, even for activities that involve some group work and applies to both undergraduate as well as graduate students. For additional requirements for graduate students, see below. All of the assessments for this course are made during class meetings. <strong>Please note:</strong> every class meeting is worth <strong>20%</strong> of your grade. Thus, you can not pass this class without attending class meetings. Grades will be calculated using the following scale, after conversion of the total number of points acquired to percentages:</p>
<p><strong>A:</strong> ≥ 90%    <strong>B:</strong> 80-89%    <strong>C:</strong> 70-79%    <strong>D:</strong> 60-69%    <strong>F:</strong> &lt; 60%</p>
<h4 id="grading-summary">5.1 Grading summary:</h4>
<hr>
<table>
<thead>
<tr class="header">
<th style="text-align: left;">Grading Component</th>
<th style="text-align: left;">Undergraduate</th>
<th style="text-align: left;">Graduate</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;">Assignments (weekly tutorials)</td>
<td style="text-align: left;">70</td>
<td style="text-align: left;">45</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;">Participation (ind &amp; group)</td>
<td style="text-align: left;">30</td>
<td style="text-align: left;">30</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;">Research Project</td>
<td style="text-align: left;">N/A</td>
<td style="text-align: left;">25</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;"><strong>Total</strong></td>
<td style="text-align: left;"><strong>100</strong></td>
<td style="text-align: left;"><strong>100</strong></td>
</tr>
</tbody>
</table>
<p><br><br></p>
<p><strong>Graduate Students</strong> will choose a research data analysis project with instructors to be completed before November 30, 2018.</p>
<h4 id="grading-details">5.2 Grading details:</h4>
<p><strong>Assignments:</strong> Working with computers in biology often requires collaboration and input from others. Almost all assignments for this course will be completed during class meetings. At the end of each class meeting you will save the entire commandline session with history command.</p>
<p><strong>Participation:</strong> Students are expected to work together to foster an atmosphere of inquiry and respect. While you will be working with computers, creative-problem solving skills will be required to enter into dialogue during class meetings and to complete tasks throughout the 5 weeks.</p>
<h3 id="university-policies-and-information">6. University Policies and Information</h3>
<p><strong>Attendance Policy, and Policy for Late and Make-up work/examinations:</strong> To succeed in this course, it is essential to show up. Twenty points will be subtracted from your grade for every absence that is not University-sponsored or deemed acceptable by the instructor. University-sponsored absences are cleared through the Office of Student Life. Please discuss anticipated absences with the instructor in advance. Make-up work will only be offered for students with acceptable absences.</p>
<p><strong>Academic Honesty:</strong> The University of Wyoming is built upon a strong foundation of integrity, respect and trust. All members of the university community have a responsibility to be honest and the right to expect honesty from others. Any form of academic dishonesty is unacceptable to our community and will not be tolerated [from the University Catalog]. Teachers and students should report suspected violations of standards of academic honesty to the instructor, department head, or dean. Other University regulations can be found <a href="http://www.uwyo.edu/generalcounsel/new-regulatory-structure/index.html">here</a>.</p>
<p><strong>Disability Statement:</strong> If you have a physical, learning, sensory or psychological disability and require accommodations, please let the instructor know as soon as possible. You will need to register with, and provide documentation of your disability to University Disability Support Services (UDSS) in SEO, room 330 Knight Hall.</p>
<p><strong>UW Writing Center:</strong> The Writing Center in Coe 302 is able to help writers at any stage of the writing process. With a focus on teaching and learning, the Writing Center is not a “fix-it shop”, but they help writers identify, articulate, and implement improvements and corrections to their writing. Drop in to see if a consultant is available, or schedule an appointment online <a href="http://uwyo.edu/writingcenter">uwyo.edu/writingcenter</a>.</p>
<h3 id="semester-schedule">7. Semester Schedule:</h3>
<p>Please note that with the exception of our first meeting, other events are TENTATIVELY scheduled, because we will work at a pace that is appropriate for the cohort.</p>
<p><strong>Module 1: Course introduction, command line interface</strong></p>
<p><em>Week 1</em></p>
<ul>
<li>Introductions, discuss survey results and syllabus</li>
<li>Lecture: Introduction to informatics</li>
<li>Introduction to Linux CLI, text editors and file manipulation. Introduction to scripting.</li>
</ul>
<p><strong>Module 2: Computer cluster, sequence similarity searches</strong></p>
<p><em>Week 2</em></p>
<ul>
<li><p>Connecting to a remote server (Mt. Teton)</p></li>
<li><p>Introduction to NCBI and various databases</p></li>
<li><p>Understanding the blast output</p></li>
<li><p>Novel genes &amp; other databases</p></li>
<li><p>Run blast on the command line</p></li>
</ul>
<p><strong>Module 3: Sequence alignments and phylogenetics</strong></p>
<p><em>Week 3</em></p>
<ul>
<li><p>Introduction to R</p></li>
<li><p>Lecture: alignments and phylogenetics</p></li>
<li><p>Run sequence alignments</p></li>
<li><p>Visualize the phylogenetic tree based on alignments</p></li>
</ul>
<p><strong>Module 4: Next-generation sequencing and quality assessment of human genomic data</strong></p>
<p><em>Week 4</em></p>
<ul>
<li><p>Lecture: Introduction to nextgen sequencing</p></li>
<li><p>Perform quality assessment on raw sequence data</p></li>
<li><p>Perform quality trimming</p></li>
<li><p>Repeat assessment to verify quality</p></li>
</ul>
<p><strong>Module 5: Analysis of Human Genomic Variation Across Global Populations</strong></p>
<p><em>Week 5</em></p>
<ul>
<li><p>Lecture: Human population genomics</p></li>
<li><p>Study SNP variant data from 1000 Genomes Project</p></li>
<li><p>Estimate allele frequencies for specific genetic loci</p></li>
<li><p>Use R programming environment for plotting data of allele frequency differences between global human populations</p></li>
</ul>
<hr>
</body>
</html>
